app/models/user.rb

```diff
@@ -1,6 +1,18 @@
 class User < ActiveRecord::Base
   has_secure_password
 
+  before_validation :strip_whitespace
   validates :first_name, :last_name, :email, presence: true
   validates :email, uniqueness: true
+  validates :password, length: { minimum: 8 }
+  validates_format_of :email,
+    with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i
+
+  private
+
+  def strip_whitespace
+    self.first_name = self.first_name.strip if self.first_name
+    self.last_name = self.last_name.strip if self.last_name
```

Comment:
I think if you use protected instead of private it may work this way

protected
def strip_whitespace
  self.first_name  = first_name.strip if first_name
  self.last_name  = last_name.strip if last_name
  self.email          = email.strip if email
end

This should work